scripts/lib/browser_client_executable.py

```
@@ -0,0 +1,245 @@
+"""Minimal JavaScript executable-range filtering for bundled Browser patch anchors."""
+
+_REGEX_PREFIX_KEYWORDS = {
+    "await", "break", "case", "continue", "debugger", "default", "delete", "do",
+    "else", "extends", "in", "instanceof", "new", "of", "return", "throw",
+    "typeof", "void", "yield",
+}
+_CONTROL_PAREN_KEYWORDS = {"catch", "for", "if", "switch", "while", "with"}
+
+
+def executable_offsets(source):
```

Ruff (0.16.1)

[warning] Too many branches (52 > 12)

(PLR0912)